### ISSUE TITLE
can't monitor the search accuracy of representation during learning

### ISSUE THREAD

**Issue by matchyc** [Shekhale/gbnns_dim_red#1]
Hi Shekhale,
I'm trying to run the code for reproducing your impressive paper results, but I got into trouble when I ran this code.
I can successfully finish training step. But when it comes to arg "val_frep" > 0, it will call cpp code (.so) to monitor the search accuracy of new representation, so I compile cpp files under the instruction `README` in `wrap/` folder.
There are some errors:
1. there is no function called `load_edge` (c_support.cpp, line 381) and `FindGraphAverageDegree`. (Even there are functions named `loadEdge` and `findGraphaverageDegree`, the parameters are totally incompatible.)
2. The last parameters of function `hnswlikeGD` was lost.

I solve these compile errors by the following steps:
1. I found that there are functions named `load_edge`  & `FindGraphAverageDegree` in another repository(Shekhale/gbnns_theory), I assumed two repositories are came from the same paper, so I copy both of them into file `c_support.cpp`. Problem 1 solved.
2. The lost parameter is `bool need_constant_degree`, so I set it to `false`. Problem 2 solved.

After that, I can compile in `wrap` folder, then get a dynamic library (.so file), and also import in python successfully.

But when I run `train.py` again, after finishing epochs, I got segmentation fault in function `hnswlikeGD`. 

I try to reproduce your impressive result in ICML 2020 paper and try to apply it for further research, so please give me some suggestions! Thank you a lot. @Shekhale 

(I modified all file paths to fit my server, I think that's will not make segmentation fault)

train command:
`python train.py --database sift --method triplet --dout 32 --epochs 4 --batch_size 512` (small epoch just for test, I use small dataset too)

error output:
```
epoch 3, times: [hn 0.49 s epoch 0.71 s val 4.40 s] lr = 0.010000 loss = -0.0224372 = 0.0117063 + lam * -3.41435, offending 1983
Computing nearest neighbors (torch)
  NN search done in 1.03 s
9900 10000 100 128 32
Loading data from /home/cm/data/sift/sift_base_valid.fvecs
knn_low 32715
Segmentation fault (core dumped)
```
